cg/meta/workflow/fluffy.py

```diff
@@ -129,82 +131,151 @@ def get_sample_control_status(self, sample_id: str) -> bool:
         sample_obj: models.Sample = self.status_db.sample(sample_id)
         return bool(sample_obj.control)
 
-    def add_concentrations_to_samplesheet(
-        self, samplesheet_housekeeper_path: Path, samplesheet_workdir_path: Path
+    def get_nr_of_header_lines_in_sample_sheet(
+        self,
+        sample_sheet_housekeeper_path: Path,
+    ) -> int:
+        """
+        Return the number of header lines in a Fluffy sample sheet.
+        Any lines before and including the line starting with [Data] is considered the header.
+
+        Returns:
+        int: The number of lines before the SAMPLE_SHEET_DATA_HEADER
+        """
+        with sample_sheet_housekeeper_path.open("r") as read_obj:
+            csv_reader = reader(read_obj)
+            header_line_count: int = 1
+            for line in csv_reader:
+                if SAMPLE_SHEET_DATA_HEADER in line:
+                    break
+                header_line_count += 1
+        return header_line_count
+
+    def read_sample_sheet_data(self, sample_sheet_housekeeper_path: Path) -> pd.DataFrame:
+        """
+        Read in a sample sheet starting from the SAMPLE_SHEET_DATA_HEADER.
+
+        Args:
+            sample_sheet_housekeeper_path (Path): Path to the housekeeper sample sheet file
+
+        Returns:
+            pd.DataFrame: A pandas dataframe of the sample sheet
+        """
+        header_line_count: int = self.get_nr_of_header_lines_in_sample_sheet(
+            sample_sheet_housekeeper_path=sample_sheet_housekeeper_path
+        )
+        return pd.read_csv(sample_sheet_housekeeper_path, index_col=None, header=header_line_count)
+
+    def add_sample_sheet_column(
+        self, sample_sheet_df: pd.DataFrame, new_column: str, to_add: list
+    ) -> pd.DataFrame:
+        """Add columns to the sample sheet.
+        Returns:
+            pd.DataFrame: Sample sheet pd.DataFrame.
+        """
+        try:
+            sample_sheet_df[new_column] = to_add
+        except ValueError:
+            LOG.error(
+                f"Error when trying to add the column: {new_column} to sample sheet with data: {to_add}."
+            )
+        return sample_sheet_df
+
+    def get_column_alias(self, sample_sheet_df: pd.DataFrame, alias: str, alternative: str):
+        """Determine column alias from the sample sheet or set to alternative.
```

Comment:
Should not these two effects (get or set alternative) be separated to make the function pure and not have any side effects?

Reply:
I made it into two separate functions, one that checks it is exists that is called by the set function